File: tournaments/isLuckyNumber/isLuckyNumber.cpp
bool isLuckyNumber(int n) {
  while (n > 0) {
    int tmpDigit = n % 10;
    if (tmpDigit != 4 && tmpDigit != 7) {
      return false;
    }
    n = n / 10;
  }
  return true;
}

File: tournaments/isLuckyNumber/isLuckyNumber.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub isLuckyNumber {
    my ($n) = @_;

    while (0 < $n) {
        if (4 != $n % 10 && 7 != $n % 10) {
            return 0;
        }
        $n = int($n / 10);
    }

    return 1;
}
